src/System.Device.Gpio.Tests/GpioControllerTestBase.cs

```diff
@@ -358,6 +403,105 @@ public void WaitForEventFallingEdgeTest()
+                Assert.True(task.Wait(TimeSpan.FromSeconds(30))); // Should end long before that
+                tokenSource.Dispose();
+            }
+        }
+
+        [Fact]
+        public void FastInterruptHandling()
+        {
+            const int numPulses = 100;
+            // These are in ms
+            const int pulseLength = 1;
+            const int waitTime = 20;
+            const double acceptableTimeFactor = 2.0;
+
+            int numInterrupts = 0;
+            int numRisingEdges = 0;
+            using (GpioController controller = new GpioController(GetTestNumberingScheme(), GetTestDriver()))
+            {
+                controller.OpenPin(InputPin, PinMode.Input);
+                controller.OpenPin(OutputPin, PinMode.Output);
+                controller.Write(OutputPin, PinValue.Low);
+                while (controller.Read(InputPin) == PinValue.High)
+                {
+                }
+
+                controller.RegisterCallbackForPinValueChangedEvent(InputPin, PinEventTypes.Rising, Callback);
+                // Ensure the poll thread is ready before we continue the actual loop
+                // If the triggers are generated by external hardware (which is what the interrupt triggering
+                // is for), we don't know when the pulses start, so no synchronisation possible there anyway
+                Thread.Sleep(10);
+                Stopwatch w = Stopwatch.StartNew();
+                for (int i = 0; i < numPulses; i++)
+                {
+                    controller.Write(OutputPin, PinValue.High);
+                    Thread.Sleep(TimeSpan.FromMilliseconds(pulseLength));
+                    controller.Write(OutputPin, PinValue.Low);
+                    Thread.Sleep(TimeSpan.FromMilliseconds(waitTime));
+                }
+
+                TimeSpan elapsed = w.Elapsed;
+                Thread.Sleep(10);
+                controller.UnregisterCallbackForPinValueChangedEvent(InputPin, Callback);
+                // All pulses must be marked as rising edge
+                Assert.Equal(numInterrupts, numRisingEdges);
+                Assert.True(numInterrupts >= numPulses - 1, $"Expected at least {numPulses - 1}, got only {numInterrupts}"); // Allow one missing pulse
+                // That's how long this test should last (at most).
+                // It will usually be around 3s, but on Windows it's around 4s (for reasons unknown)
+                double expectedMillis = numPulses * (pulseLength + waitTime);
+                Assert.True(elapsed < TimeSpan.FromMilliseconds(expectedMillis * acceptableTimeFactor), $"Used {w.Elapsed.TotalMilliseconds}ms, expected max {expectedMillis * acceptableTimeFactor}ms");
+                // Write the state of the test to the log, even if it passed (for detailed analysis)
+                _testOutputHelper.WriteLine($"Got {numInterrupts} out of {numPulses} expected. Expected a duration of {expectedMillis}, actually used {elapsed.TotalMilliseconds}ms");
```

Comment:
This should be somewhere in the output, but it looks like the build log is only available if the build failed. Any idea how to find this line (for analysis) in a succeessful build?

Reply:
Hey @pgrawehr sorry I just saw this. You can get this from here https://helix.dot.net/swagger/ui/index#!/WorkItem/WorkItem_ConsoleLog all you will need is the jobId which you should be able to get by looking at your build and search the Send To Helix step, and the logs for that should contain the GUID for job id listed. The id is always "System.Device.Gpio.Tests.dll" so by filling out those two parameters, you should be able to get the console output. Do keep in mind that we have a retention policy on the infrastructure so we won't have the logs for old builds.

Reply:
@joperezr Thanks, I'll have a look (I can always force a new build by pushing a dummy commit)

Reply:
It's a bit manual work to get the stuff I need, but I found it. The output is in the testresult.xml file, whose path can be found from the https://helix.dot.net/swagger/ui/index#!/WorkItem/WorkItem_Details section. 

It shows that for whatever reason, this test takes about 2100ms on Linux, but around 3800ms on Windows. I have no clue why this is so. Why would Thread.Sleep() be so inaccurate on windows?

Reply:
@krwq This is probably something you should once have a look into: It looks like the Windows tests are generally significantly slower than the linux tests. For instance, the relatively simple `PinValueReadAndWrite` test takes between 1ms and 12ms (depending on driver) on linux, but 23ms on Windows. I'll put this up as a separate ticket, as it is not related to this PR.

Reply:
Opened #1004